### PR TITLE
Tracker and AnonymousTracker

### PR DESCRIPTION
**Pull request by renra**
Two small helpers that remember the `userId` / `anonymousId` and inject it into other calls on the client.

More code will follow which will put them into the npm module too.

### CODE REVIEW

File: src/anonymous_tracker.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+var getArgs = require('./utils').getArgs;
+
+var publicMethods = {
+  init: function(client, anonymousId) {
+    var userId = userId;
+
+    return {
+      anonymousTrack: function() {
+        var args = getArgs(anonymousId, arguments);
+        return client.anonymousTrack.apply(this, args);
```
Comment (renra) [inline src/anonymous_tracker.js]: TODO: is there a way to delegate the method without repeating its name? Looks error-prone.

Comment (turboMaCk) [inline src/anonymous_tracker.js]: Don't know about any simple solution to that...

Comment (renra) [inline src/anonymous_tracker.js]: I think I have something - it is still going to be duplicated, but at least not triplicated 😄

**Comment by renra** [comment]
**RDY**

**Comment by turboMaCk** [comment]
We can discuss it tomorrow but this doesn't solve biggest concern which is what should happen when `track` is called before identify. Just adding state can be simple but this is not how we can use it due to concurrency. What is biggest concern? Making this change non breaking?

**Comment by renra** [comment]
Got it, tomorrow it is 🛏

**Comment by renra** [comment]
@turboMaCk Would this help your case?
https://github.com/GlobalWebIndex/segment/pull/4/commits/b93dc7c6b9ebdb925f401405094e97c90e3122ea
https://github.com/GlobalWebIndex/segment/pull/4/commits/c0b0416a9d3e29c0980eb02d67be71d109f655ec

**Comment by renra** [comment]
It's pretty much what we have on the backend, but it calls identify only once. I was also thinking - since we're about to put profile management into PRO - if the user changes his email, you'll have to run `identify` again. This could be (with this version of the code) achieved either by creating a new tracker or by adding some method that resets the `isIdentified` variable. Your call here. I don't know what will be easier to do in ember.

**Comment by renra** [comment]
Btw to make it clear, I'm fine with doing things your way. I'm not worried about breaking some api. It's just that this way it's much easier for me to understand the code. So take it just as an alternative. I don't have the ember POV so I'm probably missing stuff.

**Comment by renra** [comment]
Also I'm a bit worried whether intercom is still going to work if we don't use their library directly anymore 😬 

https://segment.com/docs/spec/identify/

**Comment by turboMaCk** [comment]
sorry I'm solving and will be solving issues for some time, I don't have time for this now.

**Comment by renra** [comment]
Sure thing. I'm going to test this in the browser in the evening and let you know. I know we want this ASAP but I say let's do it in a robust way so that we don't need to come back to it anymore. Even if it takes some more time.

**Comment by renra** [comment]
Closing in favour of: https://github.com/GlobalWebIndex/segment/pull/3

**Pull request closed without merging**